### PR TITLE
Recipe ReplaceLocalizedStreamMethods

### PR DESCRIPTION
**Pull request by BhavanaPidapa**
## What's changed?
This PR contains recipe - org.openrewrite.java.migrate.ReplaceLocalizedStreamMethods
Rule:
image

## What's your motivation?
This custom recipe replaces the occurrences of Runtime.getLocalizedInputStream and Runtime.getLocalizedOutputStream methods. The occurrence of **getLocalizedInputStream**(in) is replaced with **in** parameter and **getLocalizedOutputStream**(out) is replaced with **out** parameter.

## Anyone you would like to review specifically?
@timtebeek @cjobinabo 

## Have you considered any alternatives or workarounds?
I created sample files, created a local plugin using
./gradlew publishToMavenLocal then ran it in the Java8 Sample App using mvn rewrite:dryRun

Attaching the rewrite.patch file
[rewrite.patch](https://github.com/user-attachments/files/16617170/rewrite.patch)

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/test/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethodsTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                      }
+                  }
+                  """
+              )
+            );
+        }
+
+@Test
+void replaceGetLocalizedOutputStream() {
+    rewriteRun(
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethodsTest.java:71]: The formatting is a little off here; would you mind applying the IntelliJ defaults?

File: src/test/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethodsTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import static org.openrewrite.java.Assertions.java;
+
+public class ReplaceLocalizedStreamMethodsTest implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new ReplaceLocalizedStreamMethods("com.test.Runtime1 getLocalizedInputStream1(java.io.InputStream)", "com.test.Runtime1 getLocalizedOutputStream1(java.io.OutputStream)"));
+    }
+
+    //language=java
+    String LocalizedStreamMethodsClass = """
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethodsTest.java:17]: When a class only supports the tests but should not itself be evaluated by recipes, then it's clearer to use the `.dependsOn` you can see applied here:
https://github.com/openrewrite/rewrite-spring/blob/729164fd713cbd0bfce44e582b0d613f4d1965b0/src/test/java/org/openrewrite/java/spring/test/SpringRulesToJUnitExtensionTest.java#L34****

File: src/main/resources/META-INF/rewrite/java-version-11.yml [hunk 3]
```diff
@@ -73,6 +73,11 @@ recipeList:
       getPeerMethodPattern: java.awt.* getPeer()
       lightweightPeerFQCN: java.awt.peer.LightweightPeer
   - org.openrewrite.scala.migrate.UpgradeScala_2_12
+  - org.openrewrite.java.migrate.RemovedLogManagerPropertyChangeListenerMethods
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/java-version-11.yml:76]: Let's remove this line that carried over from
- https://github.com/openrewrite/rewrite-migrate-java/pull/526

Comment (timtebeek) [reply src/main/resources/META-INF/rewrite/java-version-11.yml:76]:
```suggestion
```

**Review by timtebeek — APPROVED** [review]
Thanks @BhavanaPidapa ! I've done a few quick polish commits but with that we're good from my side. @cjobinabo can do the final review & merge.

File: src/main/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethods.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            @Nullable String localizedOutputStreamMethodMatcher) {
+        this.localizedInputStreamMethodMatcher = localizedInputStreamMethodMatcher == null ?
+                "java.lang.Runtime getLocalizedInputStream(java.io.InputStream)" : localizedInputStreamMethodMatcher;
+        this.localizedOutputStreamMethodMatcher = localizedOutputStreamMethodMatcher == null ?
+                "java.lang.Runtime getLocalizedOutputStream(java.io.OutputStream)" : localizedOutputStreamMethodMatcher;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Replace getLocalizedInputStream and getLocalizedOutputStream with direct assignment";
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethods.java:58]:
```suggestion
        return "Replace `getLocalizedInputStream` and `getLocalizedOutputStream` with direct assignment";
```

File: src/main/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethods.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Replace `getLocalizedInputStream` and `getLocalizedOutputStream` with direct assignment";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace `Runtime.getLocalizedInputStream(InputStream)` and `Runtime.getLocalizedOutputStream(with the argument)` with their direct arguments.";
```
Comment (cjobinabo) [inline src/main/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethods.java:63]: In the description, I think it would be helpful to add some info about why this change is made. That the earlier implementation of these methods simply returned the input parameter.

Comment (BhavanaPidapa) [reply src/main/java/org/openrewrite/java/migrate/ReplaceLocalizedStreamMethods.java]: Updated the description as mentioned.

**Review by cjobinabo — APPROVED** [review]
